### DISCUSSION TITLE
Any issues with calling receiveDirect() as part of a loop?

### DISCUSSION THREAD

**Discussion by Crsarmv7l** [jgromes/RadioLib#1090] [Q&A]
Direct Mode reads when clock is high. What I am wondering is can I only start/stop reading when clock is high based on RSSI.

The below is concept code that would be in a loop, with RSSIFlag indicating that there is an RSSI param entered by me, and RSSI being the desired RSSI floor. This is rough, I can see a few place I will have issues as is.

So many edits: Worked it out a bit more, I'll probably just end up testing once I get home.
```
if (RSSIFlag) {	
        radio.receiveDirect();
	while (RSSI =< radio.getRSSI()) {
	    if (radio.available()=1) {
		byte b = radio.read();
		Serial.print(b, HEX);
	     }
        }
        radio.packetMode();
	Serial.println();
}
else {
	radio.receiveDirect();
	if (radio.available() >= 60) {
		while (radio.available()) {
			byte b = radio.read();
			Serial.print(b, HEX);
		}
		Serial.println();
	}
        radio.packetMode();
}
```

**Comment by jgromes** [comment]
I'm not entirely sure what are you trying to accomplish - only reading data when RSSI is above some threshold? If so, then keep in mind reading the RSSI is an SPI transaction and that takes time.

> Direct Mode reads when clock is high. 

Not exactly, it samples the data pin when it detects rising edge on clock pin.

**Comment by Crsarmv7l** [comment]
Yes the goal is to only read data when RSSI is above a certain threshold.

I'm certainly open to suggestions

**Reply by jgromes** [reply]
I think that the way to go about this would be:

1. Add a flag to the ISR that will not allow data to be read if the flag is not set
2. In loop, call getRSSI as fast as possible
3. Once the RSSI exceeds some threshold, set the flag and stop checking RSSI
4. After some time (e.g. pre-determined number of received bytes), reset the flag and start checking RSSI again

This will essentially try to determine the start of packet based on RSSI threshold, as opposed to the the default approach in direct mode which is sync word.

**Reply by Crsarmv7l** [reply]
Hmm I see. I think I will have to play with this and see what I can workout. Most of the signals I have seen/been interested in are at a max of ~19,200 baud but usually lower. Using SPI might not be much of a bottleneck but we will see.

For sure sync word is the more appropriate method, but sometimes you don't know the signal syncword.

Thanks for another method to explore!
